spring-kafka/src/main/java/org/springframework/kafka/core/KStreamBuilderFactoryBean.java

```diff
@@ -116,8 +130,12 @@ public synchronized void start() {
 		if (!this.running) {
 			try {
 				this.kafkaStreams = new KafkaStreams(getObject(), this.streamsConfig, this.clientSupplier);
-				this.kafkaStreams.setStateListener(this.stateListener);
-				this.kafkaStreams.setUncaughtExceptionHandler(this.exceptionHandler);
+				if (this.stateListener != null) {
```

Comment:
??? The `fb.start()` creates a new `KafkaStreams` each time.
So, that `getKafkaStreams().setStateListener(myStateListener)` is irrelevant.
WDYT?